### ISSUE TITLE
Reading Order (IMPACT)

### ISSUE THREAD

**Issue by markusenders** [altoxml/schema#18]
use case
Modern OCR software is able to recognize sections within a page. The logical text flow of between some of the sections may be continuous. The OCR software however may not always be able recognize the text flow correctly and stores these sections in non-continuous parts of the ALTO file.

Additional processing software or manual intervention may add correct this problem. ALTO has to store the reading order explicitly. The reading order should not rely on the order of XML elements in the ALTO file.

implementation
A single <ReadingOrder> element defines the information flow for every section in the document. This section is called Region. Each region is specified by a <RegionRef> element. This element points to either one Block-element (see chapter 4). Each region is part of a group.  A group can contain regions that are

•unordered (information flow doesn’t have a particular order) or
•ordered (information flow has a particular order). 
The appropriate elements are called <OrderedGroup> or <UnorderedGroup>. Every region must be part of exactly one group. All regions in the ordered group must provide their position within the group. This position is stored in the ORDER attribute. The value of the ORDER attribute must be an integer and be unique within the group.

In order to represent complex information flows within a page groups may have an unlimited number of sub-groups. The sub-groups are of the type ordered and unordered groups. Both types of groups may have any type of sub-groups.

example

```
       <ReadingOrder>

               <OrderedGroup ID=”G1”>

                     <RegionRef IDREF="xxxx001" ORDER="1"/>

                     <RegionRef IDREF="…" ORDER="2"/>

                     <RegionRef IDREF="……" ORDER="3"/>

               </OrderedGroup>

               <UnorderedGroup ID=”UG1!”>

                     <RegionRef IDREF="…"/>

                     <RegionRef IDREF="……"/>

               </UnorderedGroup>

      </ReadingOrder>

      <Layout>

                     <Page>

                           <PrintSpace>

                                 <TextBlock ID="xxxx001">

                                       <TextLine>

                                             <String CONTENT="Advertisement"/>

                                       </TextLine>

                                 </TextBlock>

                           </PrintSpace>

                     </Page>          

                           ..... the complete layout description

   </Layout>
```

**Comment by cowboyMontana** [comment]
Changed label from 'submitted' to 'discussion'.

**Comment by cowboyMontana** [comment]
Assigned Markus Enders as change request champion.

**Comment by artunit** [comment]
As per the [2021-04-29 Board Meeting](/altoxml/board/blob/gh-pages/minutes/2021/2021-04-29%20ALTO%20Board%20Meeting%20Minutes.md), the _High Priority_ label will moved to [Clarify implicit reading order #68](https://github.com/altoxml/schema/issues/68) to reflect a new focus on this issue.

**Comment by cneud** [comment]
ACCEPT

**Comment by cowboyMontana** [comment]
ACCEPT

**Comment by cowboyMontana** [comment]
ACCEPT

**Comment by artunit** [comment]
ACCEPT

**Comment by jukervin** [comment]
ACCEPT

**Comment by callylaw** [comment]
ACCEPT

**Comment by Haighton** [comment]
ACCEPT

**Comment by Ra1phM** [comment]
ACCEPT

**Comment by ntra00** [comment]
ACCEPT

**Comment by JLoitzenbauer-CRKN** [comment]
ACCEPT

**Comment by SebastienC-BnF** [comment]
ACCEPT

**Comment by rajubln** [comment]
ACCEPT

**Comment by acpopat** [comment]
ACCEPT

**Comment by cipriandinu** [comment]
Published on v4.3
